lib/compress/zstd_compress.c

```diff
@@ -2240,7 +2245,7 @@ static ZSTD_blockCompressor ZSTD_selectBlockCompressor(ZSTD_strategy strat, int
 
 static size_t ZSTD_compressBlock_internal(ZSTD_CCtx* zc, void* dst, size_t dstCapacity, const void* src, size_t srcSize)
 {
-    ZSTD_blockCompressor const blockCompressor = ZSTD_selectBlockCompressor(zc->params.cParams.strategy, zc->lowLimit < zc->dictLimit);
+    ZSTD_blockCompressor const blockCompressor = ZSTD_selectBlockCompressor(zc->params.cParams.strategy, zc->seqStore.forceExtDict || (zc->lowLimit < zc->dictLimit));
```

Comment:
I thought also about:
1) creating hidden `ZSTD_compressBlock_internal_extDict` with `extDict=1` but it forces to create plenty of other hidden functions.
2) creating a new function only to test `ZSTD_selectBlockCompressor` with `extDict=1`